aea/connections/base.py

```diff
@@ -63,15 +69,18 @@ def __init__(
         parameters are None: connection_id, excluded_protocols or restricted_to_protocols.
 
         :param configuration: the connection configuration.
-        :param address: the address.
+        :param identity: the identity object held by the agent.
+        :param cryptos: the crypto store for encrypted communication.
         :param restricted_to_protocols: the set of protocols ids of the only supported protocols for this connection.
         :param excluded_protocols: the set of protocols ids that we want to exclude for this connection.
         :param connection_id: the connection identifier.
         """
         super().__init__(configuration)
-        self._loop = None  # type: Optional[AbstractEventLoop]
+        self._loop: Optional[AbstractEventLoop] = None
         self._connection_status = ConnectionStatus()
-        self._address = address  # type: Optional[Address]
+
+        self._identity: Optional[Identity] = identity
+        self._cryptos: CryptoStore = cryptos if cryptos is not None else CryptoStore()
```

Comment:
why not `Optional[CryptoStore] = cryptos`?

Reply:
It was to simplify the constructor, so you don't have to necessarily provide the crypto store, and calls to self.cryptos will work. But I don't mind adding the assert (below)